test_runner/src/main/kotlin/ftl/util/OutcomeDetailsFormatter.kt

```diff
@@ -0,0 +1,76 @@
+package ftl.util
+
+import com.google.api.services.toolresults.model.FailureDetail
+import com.google.api.services.toolresults.model.InconclusiveDetail
+import com.google.api.services.toolresults.model.Outcome
+import com.google.api.services.toolresults.model.SkippedDetail
+import ftl.reports.api.data.TestSuiteOverviewData
+import ftl.util.StepOutcome.failure
+import ftl.util.StepOutcome.flaky
+import ftl.util.StepOutcome.inconclusive
+import ftl.util.StepOutcome.skipped
+import ftl.util.StepOutcome.success
+import ftl.util.StepOutcome.unset
+
+fun Outcome.getDetails(testSuiteOverviewData: TestSuiteOverviewData?) = when (summary) {
+    success, flaky -> testSuiteOverviewData?.let {
+        getSuccessOutcomeDetails(
+            testSuiteOverviewData = it,
+            otherNativeCrash = successDetail?.otherNativeCrash ?: false
+        )
+    }
+    failure -> failureDetail.getFailureOutcomeDetails(testSuiteOverviewData)
+    inconclusive -> inconclusiveDetail.formatOutcomeDetails()
+    skipped -> skippedDetail.formatOutcomeDetails()
+    unset -> "unset"
+    else -> "unknown"
+}
+
+private fun getSuccessOutcomeDetails(
+    testSuiteOverviewData: TestSuiteOverviewData,
```

Comment:
If test `TestSuiteOverviewData` will be receiver this method, will looks:
```
private fun TestSuiteOverviewData.getSuccessOutcomeDetails(
    otherNativeCrash: Boolean
) = StringBuilder("$successCount test cases passed").apply {
    if (skipped > 0) append(skippedMessage(skipped))
    if (flakes > 0) append(flakesMessage(flakes))
    if (otherNativeCrash) append(NATIVE_CRASH_MESSAGE)
}.toString()
```
and `getDetails` also will be shorter:
```
fun Outcome.getDetails(testSuiteOverviewData: TestSuiteOverviewData?) = when (summary) {
    success, flaky -> testSuiteOverviewData?.getSuccessOutcomeDetails(successDetail?.otherNativeCrash ?: false)
    failure -> failureDetail.getFailureOutcomeDetails(testSuiteOverviewData)
    inconclusive -> inconclusiveDetail.formatOutcomeDetails()
    skipped -> skippedDetail.formatOutcomeDetails()
    unset -> "unset"
    else -> "unknown"
}
```